public/client/TracyDebug.hpp

```diff
@@ -1,9 +1,25 @@
 #ifndef __TRACYPRINT_HPP__
 #define __TRACYPRINT_HPP__
 
+#ifdef TRACY_ON_DEMAND
+#  define TRACY_VERBOSE_EARLY_OUT_COND if( !GetProfiler().IsConnected() ) break
+#else
+#  define TRACY_VERBOSE_EARLY_OUT_COND assert( tracy::ProfilerAvailable() )
+#endif
+
+#define TracyInternalMessage( severity, ... )																				   \
+	do {																													   \
+        TRACY_VERBOSE_EARLY_OUT_COND;																						   \
+		char buffer[4096];																									   \
+		snprintf( buffer, sizeof(buffer), __VA_ARGS__ );																	   \
+		tracy::Profiler::LogString( tracy::MessageSourceType::Tracy, severity, 0, TRACY_CALLSTACK, strlen( buffer ), buffer ); \
+	} while( 0 )
+
 #ifdef TRACY_VERBOSE
 #  include <stdio.h>
-#  define TracyDebug(...) fprintf( stderr, __VA_ARGS__ );
+#  define TracyDebug(...) do { fprintf( stderr, __VA_ARGS__ ); fputc( '\n' ); } while( 0 )
+#elif !defined(TRACY_NO_INTERNAL_MESSAGE)
```

Comment:
Thinking ahead here, but if we decide to remove `TRACY_NO_INTERNAL_MESSAGE` we could have `TracyInternalMessage` defer the formatted message to a special TracyInternalLog function where we can have more actions (like always printing to stderr if it is an internal error, capturing callstacks, etc, to make our life debugging Tracy easier).